plugins/modules/falconctl.py

```diff
@@ -0,0 +1,372 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Ansible module to configure CrowdStrike Falcon Sensor on Linux systems.
+# Copyright: (c) 2021, CrowdStrike Inc.
+
+# Unlicense (see LICENSE or https://www.unlicense.org)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: falconctl
+author:
+  - Gabriel Alford <redhatrises@gmail.com>
+  - Carlos Matos <matosc15@gmail.com>
+short_description: Configure CrowdStrike Falcon Sensor
+description:
+  - Configures CrowdStrike Falcon Sensor on Linux systems
+options:
+    state:
+      description:
+        - Ensures that requested parameters are removed (absent) or added (present) to the Falcon sensor.
+      type: str
+      required: yes
+      choices: [ absent, present ]
+    cid:
+      description:
+        - CrowdStrike Falcon Customer ID (CID).
+      type: str
+    provisioning_token:
+      description:
+        - Installation tokens prevent unauthorized hosts from being accidentally or maliciously added to your customer ID (CID).
+        - Optional security measure for your CID.
+        - This paramter requires supplying a C(cid).
+      type: str
+    aid:
+      description:
+        - Whether or not you would like to delete the associated Agent ID.
+        - Useful when preparing a host as a master image for cloning or virtualization.
+        - This applies only to C(state=absent).
+      type: bool
+    apd:
+      description:
+        - Whether to enable or disable the Falcon sensor to use a proxy.
+      type: bool
+    aph:
+      description:
+        - Specifies the application proxy host to use for Falcon sensor proxy configuration.
+      type: str
+    app:
+      description:
+        - Specifies the application proxy port to use for Falcon sensor proxy configuration.
+      type: int
+    trace:
+      description:
+        - Configure the appropriate trace level.
+      type: str
+      choices: [ none, err, warn, info, debug ]
+    feature:
+      description:
+        - Configure the Falcon sensor feature flags.
+      type: list
+      elements: str
+      choices: [ none, enableLog, disableLogBuffer, disableOsfm, emulateUpdate ]
+    metadata_query:
+      description:
+        - Configure the Falcon sensor cloud provider metadata query flags.
+      type: str
```

Comment:
@redhatrises @isimluk Can you guys provide better verbiage for this? Also, please double check the other descriptions and make sure they a) make sense and b) are correct lol..